From d9d61d88046a27f3149677c4fe3de2827b7dae3d Mon Sep 17 00:00:00 2001
From: Zhipeng Gong <zhipeng.gong@intel.com>
Date: Sun, 14 Jan 2001 04:12:08 +0800
Subject: [PATCH 075/143] drm/i915/bdw: Implement WaDisableFenceDestinationToSLM for GT1/2

This WA is needed for BDW E,F step, not just GT3.
---
 drivers/gpu/drm/i915/intel_ringbuffer.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Index: current/drivers/gpu/drm/i915/intel_ringbuffer.c
===================================================================
--- current.orig/drivers/gpu/drm/i915/intel_ringbuffer.c
+++ current/drivers/gpu/drm/i915/intel_ringbuffer.c
@@ -885,7 +885,7 @@ static int bdw_init_workarounds(struct i
 			  /* WaForceContextSaveRestoreNonCoherent:bdw */
 			  HDC_FORCE_CONTEXT_SAVE_RESTORE_NON_COHERENT |
 			  /* WaDisableFenceDestinationToSLM:bdw (pre-prod) */
-			  (IS_BDW_GT3(dev) ? HDC_FENCE_DEST_SLM_DISABLE : 0));
+			  ((IS_BDW_GT3(dev) || (INTEL_REVID(dev) <= 0x6)) ? HDC_FENCE_DEST_SLM_DISABLE : 0));
 
 	return 0;
 }
